Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -603,6 +603,78 @@ public class GraphTraverser: GraphTraversing {
         )
     }
 
+    public func targetsWithExternalDependencies() -> Set<GraphTarget> {
+        Set(graph.dependencies.compactMap { fromDependency, toDependencies in
+            let fromNonExternalTarget: GraphTarget?
+            switch fromDependency {
+            case let .target(targetName, projectPath):
+                let project = graph.projects[projectPath]!
+                if project.isExternal {
+                    fromNonExternalTarget = nil
+                } else {
+                    let target = graph.targets[projectPath]![targetName]!
+                    fromNonExternalTarget = GraphTarget(path: projectPath, target: target, project: project)
+                }
+            case .bundle, .framework, .library, .packageProduct, .sdk, .xcframework:
+                fromNonExternalTarget = nil
+            }
+            guard fromNonExternalTarget != nil else { return nil }
```

Comment:
```suggestion
            let fromNonExternalTarget: GraphTarget
            switch fromDependency {
            case let .target(targetName, projectPath):
                let project = graph.projects[projectPath]!
                if project.isExternal {
                    return nil
                } else {
                    let target = graph.targets[projectPath]![targetName]!
                    fromNonExternalTarget = GraphTarget(path: projectPath, target: target, project: project)
                }
            case .bundle, .framework, .library, .packageProduct, .sdk, .xcframework:
                return nil
            }
```

This way, we can skip the guard check and just return early from the `switch`